src/Features/Core/Portable/InheritanceMargin/Finders/DerivedTypeSymbolsFinder.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.FindSymbols;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.InheritanceMargin.Finders
+{
+    internal class DerivedTypeSymbolsFinder : InheritanceSymbolsFinder
+    {
+        public static readonly DerivedTypeSymbolsFinder Instance = new();
+
+        protected override async Task<ImmutableArray<ISymbol>> GetAssociatedSymbolsAsync(ISymbol symbol, Solution solution, CancellationToken cancellationToken)
+        {
+            var derivedSymbols = await InheritanceMarginServiceHelper.GetDerivedTypesAndImplementationsAsync(solution, (INamedTypeSymbol)symbol, cancellationToken).ConfigureAwait(false);
+            var indegreeSymbolMap = GetIndegreeSymbolMap(derivedSymbols);
+
+            return TopologicalSortAsArray(derivedSymbols.CastArray<ISymbol>(), indegreeSymbolMap);
+        }
+
+        private static ImmutableDictionary<ISymbol, HashSet<ISymbol>> GetIndegreeSymbolMap(ImmutableArray<INamedTypeSymbol> derivedSymbols)
```

Comment:
so immutabledicts are pretty expensive/slow.  they're good when they're going to be stored a fields (and thus you want to ensure that nothing is changing them).  consider using normal/poooled dictionary here for perf if it's just for temp data that is used as part of a computation.

Reply:
They could be normal dictionary/set. Will update it